lib/Dialect/FIRRTL/Transforms/IMDeadCodeElim.cpp

```diff
@@ -90,15 +95,12 @@ struct IMDeadCodeElimPass : public IMDeadCodeElimBase<IMDeadCodeElimPass> {
 };
 } // namespace
 
-void IMDeadCodeElimPass::markWireOrRegOrNode(Operation *op) {
-  assert(isWireOrRegOrNode(op) && "only a wire, a reg or a node is expected");
+void IMDeadCodeElimPass::markWireOrRegOrNodeOrMemory(Operation *op) {
+  assert(isWireOrRegOrMemoryNode(op) &&
+         "only a wire, a reg or a node is expected");
```

Comment:
Nit: 
```suggestion
         "only a wire, a reg, a memory, or a node is expected");
```
Hidden nit in the above: Oxford comma.